src/blocks/CarouselBlock.php

```diff
@@ -48,21 +49,62 @@ public function config()
-                ['type' => self::TYPE_IMAGEUPLOAD, 'var' => 'image', 'label' => Module::t('block_carousel.image')]
+                ['var' => 'images', 'label' => Module::t('block_carousel.items'), 'type' => self::TYPE_MULTIPLE_INPUTS, 'options' => [
+                    ['var' => 'title', 'type' => self::TYPE_TEXT, 'label' => Module::t('block_carousel.title')],
+                    ['var' => 'caption', 'type' => self::TYPE_TEXTAREA, 'label' => Module::t('block_carousel.caption')],
+                    ['var' => 'image', 'type' => self::TYPE_IMAGEUPLOAD, 'label' => Module::t('block_carousel.image')],
+                    ['var' => 'alt', 'type' => self::TYPE_TEXT, 'label' => Module::t('block_carousel.alt')],
+                    ['var' => 'link', 'type' => self::TYPE_LINK, 'label' => Module::t('block_carousel.image_link')]
+                ]],
+            ],
+            'cfgs' => [
+                ['var' => 'controls', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_controls'), 'initvalue' => 1],
+                ['var' => 'indicators', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_indicators'), 'initvalue' => 1],
+                ['var' => 'crossfade', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_crossfade'), 'initvalue' => 1],
+                ['var' => 'interval', 'type' => self::TYPE_NUMBER, 'label' => Module::t('block_carousel.config_interval')],
+                ['var' => 'keyboard', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_keyboard'), 'initvalue' => 1],
+                ['var' => 'pause', 'type' => self::TYPE_TEXT, 'label' => Module::t('block_carousel.config_pause')],
+                ['var' => 'ride', 'type' => self::TYPE_TEXT, 'label' => Module::t('block_carousel.config_ride')],
+                ['var' => 'wrap', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_wrap'), 'initvalue' => 0],
+                ['var' => 'row', 'type' => self::TYPE_CHECKBOX, 'label' => Module::t('block_carousel.config_row'), 'initvalue' => 0]
             ]
         ];
     }
-    
+
+    public function images()
+    {
+        $images = [];
+        foreach ($this->getVarValue('images', []) as $item) {
+            $images[] = [
+                'image'             => isset($item['image']) ? BlockHelper::imageUpload($item['image'], false, true) : null,
+                'alt'               => isset($item['alt']) ? $item['alt'] : 'no-alt-text-set',
+                'title'             => isset($item['title']) ? $item['title'] : '',
+                'caption'           => isset($item['caption']) ? $item['caption'] : '',
+                'link'              => isset($item['link']) ? BlockHelper::linkObject($item['link']) : null,
+            ];
+        }
+        return $images;
+    }
+
+    public function conf() {
```

Comment:
please use psr2 coding standards, maybe also rename the method to something which represents what it does like `getCarouselConfig()`